### PR TITLE
Show warning if user already set an application password

### PR DESCRIPTION
**Pull request by DL6ER**
# What does this implement/fix?

See title.

### No app password is set

Screenshot from 2024-06-15 11-23-25

### App password is already set

Screenshot from 2024-06-15 11-22-25

---

**Related issue or feature (if applicable):** N/A

**Pull request in [docs](https://github.com/pi-hole/docs) with documentation (if applicable):** N/A

---
**By submitting this pull request, I confirm the following:** 

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against. 
2. I have commented my proposed changes within the code.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

## Checklist:

- [x] The code change is tested and works locally.
- [x] I based my code and PRs against the repositories `developmental` branch.
- [x] I [signed off](https://docs.pi-hole.net/guides/github/how-to-signoff/) all commits. Pi-hole enforces the [DCO](https://docs.pi-hole.net/guides/github/dco/) for all contributions
- [x] I [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) all my commits. Pi-hole requires signatures to verify authorship
- [x] I have read the above and my PR is ready for review.

### CODE REVIEW

**Comment by yubiuser** [comment]
This is a nice idea. I only fear that the hint is not strong enough. Could we have this warning as a preceding modal that allows to cancel or proceed (to the actual app password modal) and requires an active click to continue?

**Review by rdwebdesign — DISMISSED** [review]

File: settings-api.lp [hunk 1]
```diff
@@ -260,6 +260,9 @@ mg.include('scripts/pi-hole/lua/settings_header.lp','r')
                 <p>IMPORTANT: The app password generated here will only be shown
                 once and cannot be recovered. Make sure to store it in a safe
                 place!</p>
+                <div class="danger-area" style="display: none;" id="existing_apppw_warning">
```
Comment (rdwebdesign) [inline settings-api.lp]: As an alternative to yubi's suggestion, we could use a different style for the warning box, using different CSS classes:

```suggestion
                <div class="callout callout-warning" style="display: none;" id="existing_apppw_warning">
```

---

We also need to remove lines 25 to 27 from `default-dark.css` theme _(or change its color to `#fff`)_:
https://github.com/pi-hole/web/blob/3681d7187a44b32a667f40b585629d45bf4e44e9/style/themes/default-dark.css#L25-L27

**Review by rdwebdesign — CHANGES_REQUESTED** [review]

**Comment by rdwebdesign** [comment]
Screenshots:

image

**Review by yubiuser — CHANGES_REQUESTED** [review]
On midnight and high contrast dark theme the cross is hard to see

Screenshot at 2024-06-25 22-21-39
____

Why do those two buttons have different colors? Is one more important than the other?
Screenshot at 2024-06-25 22-21-58

___

If no app password is currently set, we should not show the button to remove it.
Screenshot at 2024-06-25 22-22-40

**Comment by rdwebdesign** [comment]
### 1.
> On midnight and high contrast dark theme the cross is hard to see

Fixed.

### 2.
> Why do those two buttons have different colors? Is one more important than the other?

I don't know why they were created like this, but I don't mind the different colors in this case. Maybe someone else has an explanation for the colors.

### 3. 
> If no app password is currently set, we should not show the button to remove it.

~~I agree.~~ Actually, the app password is always generated when this modal is opened:
https://github.com/pi-hole/FTL/blob/0367117cad371bb9cbe41dfd0ead69c39e8e5192/src/api/api.c#L39

https://github.com/pi-hole/FTL/blob/0367117cad371bb9cbe41dfd0ead69c39e8e5192/src/api/2fa.c#L312-L338

**Comment by yubiuser** [comment]
> Actually, the app password is always generated when this modal is opened:

Mhh.. sure, it must be generated already to show it to the user. But is it already saved when the modal is opened?

**Comment by rdwebdesign** [comment]
> But is it already saved when the modal is opened?

I'm not sure. Maybe @DL6ER can answer this question.

**Comment by DL6ER** [comment]
> Actually, the app password is always generated when this modal is opened

> Mhh.. sure, it must be generated already to show it to the user. But is it already saved when the modal is opened?

Yes and No. It is only saved (and, hence, made active) once the user confirms this.

> Maybe someone else has an explanation for the colors.

`Enable 2FA` is green as it is "good" (as in: it enhances security, especially with very simple passwords). App passwords, on the other hand, are neutral in terms of security (as long as they are kept secret it's basically impossible to brute-force it due to its design complexity and a cryptographically secure source of randomness used to generate it).

**Review by rdwebdesign — APPROVED** [review]

**Review by yubiuser — APPROVED** [review]

**Pull request merged**